mono/profiler/proflog.c

```diff
@@ -217,20 +217,24 @@ typedef struct _LogBuffer LogBuffer;
  *
  * type metadata format:
  * type: TYPE_METADATA
- * exinfo: flags: TYPE_LOAD_ERR
+ * exinfo: one of: TYPE_END_LOAD, TYPE_END_UNLOAD (optional for TYPE_THREAD and TYPE_DOMAIN)
  * [time diff: uleb128] nanoseconds since last timing
  * [mtype: byte] metadata type, one of: TYPE_CLASS, TYPE_IMAGE, TYPE_ASSEMBLY, TYPE_DOMAIN,
- * TYPE_THREAD
+ * TYPE_THREAD, TYPE_CONTEXT
  * [pointer: sleb128] pointer of the metadata type depending on mtype
+ * [flags: uleb128] must be 0
  * if mtype == TYPE_CLASS
  *	[image: sleb128] MonoImage* as a pointer difference from ptr_base
- * 	[flags: uleb128] must be 0
  * 	[name: string] full class name
  * if mtype == TYPE_IMAGE
- * 	[flags: uleb128] must be 0
  * 	[name: string] image file name
- * if mtype == TYPE_THREAD
- * 	[flags: uleb128] must be 0
+ * if mtype == TYPE_ASSEMBLY
+ * 	[name: string] assembly name
+ * if mtype == TYPE_DOMAIN && exinfo == 0
+ * 	[name: string] domain friendly name
+ * if mtype == TYPE_CONTEXT
+ * 	[domain: sleb128] domain id as pointer
```

Comment:
It's the domain ID associated with the context (a context always belongs to one domain, while one domain can have many contexts). The "as a pointer" just means we encode it as a pointer value (even though it's a 32-bit integer) to be consistent with the other `TYPE_METADATA` events.